### PR TITLE
Dockerfile modification_automated Intel Quantum simulator launching process in a container

### PR DESCRIPTION
**Pull request by rashed2940**
Modified dockerfile will enable user quickly to spin up a container and run intel quantum simulator to test their code. 
as there is a modification in dockerfile occured, README.md file is also update accordingly. 
Added new lines in `Dockerfile` are attached below to review.
 
#### install lib for missing pthread module
RUN apt-get -y install libboost-all-dev

#### installing and configuring conda env
RUN wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda.sh
RUN bash ~/miniconda.sh -b -p $HOME/miniconda
ENV PATH="/root/miniconda/bin:$PATH"
RUN /bin/bash -c ". ~/.bashrc && \
                conda install -y notebook && \
                conda install -y pybind11 && \
                conda install -y numpy && \
                conda install -y matplotlib" 

#############################################

Attached below new lines in `README.md` file to review. 

## Docker: build image and run/execute container

`Dockerfile` includes the instructions to build the docker image of an Ubuntu machine
with Intel-QS already installed. The image can be 'run' to create a container.
The container can be 'executed' to login into the machine.
For this, Three steps need to be followed to prepare containerized intelqs simulator.

*  Create container
*  Configure SSH tunneling
*  Launch Jupyter notebook

#### Create container

Important: If user is not `root`,add `sudo` before each bash and docker command.
First command in below will build docker image for Intelqs named qhipster.One can choose other name too.
(if choosen name is other than qhipster, replace that name in all other commands as well).
Second command will create container while mapping 8080 port to the localhost.
Third command will enable conda env,execute cmake and make command, it will launch jupyter notebook
in the terminal. Copy and save the printed token from the terminal.
`Example of a given token : http://127.0.0.1:8080/?token=6ee42173ee71353c1f1b33f8feb33132aed15f2a07960bc8`.
Last command is optional,this can be executed to go inside container.

```bash
  docker build -t qhipster .
  docker run -d -t -p 8080:8080 qhipster
  docker exec -i $(docker ps|grep -i qhipster|cut -d ' ' -f1) /bin/bash -c ". ~/.bashrc && . /opt/intel/mkl/bin/mklvars.sh intel64 ilp64 && . /opt/intel/bin/compilervars.sh -arch intel64 -platform linux && mkdir build && cd build && CXX=g++ cmake -DIqsMPI=OFF -DIqsUtest=ON -DIqsPython=ON .. && make && cd .. && jupyter notebook --ip 0.0.0.0 --port 8080 --no-browser --allow-root"
  docker exec -it $(docker ps|grep -i qhipster|cut -d ' ' -f1) /bin/bash
```

If Docker is used on a Windows host machine, the last line should be substituted by:
`winpty docker exec -it <container_id> //bin/bash`.

#### Configure SSH tunneling

In local laptop, open Ubuntu emulator. This to allow the use of SSH protocol for port forwarding.
For example if you use Mobaxterm tool,launch a session and type following command in the Mobaxterm shell:

```bash
  ssh -L 8080:localhost:8080 user@domain.com
```

#### Launch Jupyter notebook

Now, paste the copied token(which we copied from 'Create Container' section) in your preferred web browser
(most importantly, please clear your browser cache before pasting the token). Once you are seeing all
folders and files in browser, follow below section to begin.

### CODE REVIEW

**Review by fbaru-dev — CHANGES_REQUESTED** [review]
Thank you for your changes. I have added some comment, which needs to be addressed before the merging.
The extensive documentation you have provided will be part of the wiki page we are implementing.

File: Dockerfile [hunk 1]
```diff
@@ -18,14 +18,14 @@
 FROM ubuntu:latest
 
 # Setup the proxy server environment.
-ENV http_proxy http://proxy-chain.intel.com:911
-ENV https_proxy http://proxy-chain.intel.com:911
+#ENV http_proxy http://proxy-chain.intel.com:911
+#ENV https_proxy http://proxy-chain.intel.com:911
```
[the remark this answers is not in the document]
Comment (rashed2940) [reply Dockerfile]: yes, i will remove these 2 lines.

File: README.md [hunk 2]
```diff
@@ -167,7 +197,7 @@ define a qubit register object, perform quantum gates, measure one or multiple q
 
 If the Python bindings were enabled, the same learning can be performed using the iPython
 notebook `tutorials/get_started_with_IQS.ipynb`.
-
+(Important note: comment line "from numpy import random_inte")
```
[the remark this answers is not in the document]
Comment (rashed2940) [reply README.md]: As "from numpy import random_intel" is failing, 
i simply installed numpy package , and then all worked fine.

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (fbaru-dev) [reply Dockerfile]: great!

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (fbaru-dev) [reply README.md]: great!

**Comment by fbaru-dev** [comment]
This is an update of the Dockerfile to provide a container solution for the simulator.

**Pull request merged**